tests/value-converters/intl-duration-format-value-converter.test.ts

```
@@ -0,0 +1,118 @@
+            years: -2
+        },
+        optionId: null,
+        expectedFormat: '730 days ago'
+    },
+    {
+        input: <DateTimeComponents>{
+            years: -2
+        },
+        optionId: "years-only",
+        expectedFormat: '2 years ago'
+    },
+    {
+        input: <DateTimeComponents>{
+            days: 3,
+            hours: 4,
+            minutes: 5,
+            seconds: 6
+        },
+        optionId: null,
+        expectedFormat: "in 3 days, in 4 hours, in 5 minutes, in 6 seconds"
+    },
+    {
+        input: <DateTimeComponents>{
+            days: -1,
+            hours: -1,
+            minutes: -1,
+            seconds: -1
+        },
+        optionId: null,
+        expectedFormat: "1 day ago, 1 hour ago, 1 minute ago, 1 second ago"
+    },
+    {
+        input: <DateTimeComponents>{
+            days: 188
+        },
+        optionId: "quarters-and-weeks",
+        expectedFormat: "in 2 quarters, in 1 week, in 1 day"
+    },
+    {
+        input: <DateTimeComponents>{
+            days: 2,
+        },
+        optionId: "day-only",
+        locale: "es",
+        expectedFormat: "pasado mañana"
+    },
+    {
+        input: <DateTimeComponents>{
+            hours: -100,
+        },
+        optionId: "days-and-hours",
+        locale: "es",
+        expectedFormat: "hace 4 días, hace 4 horas"
+    },
+    {
+        input: <DateTimeComponents>{
+            hours: -100,
+        },
+        optionId: "days-and-hours",
+        locale: "en",
+        expectedFormat: "4 days ago, 4 hours ago"
+    },
+    {
+        input: <DateTimeComponents>{
+            hours: -25,
+        },
+        optionId: "days-and-hours",
+        locale: "en",
+        expectedFormat: "yesterday, 1 hour ago"
+    },
+    {
+        input: <DateTimeComponents>{
+            seconds: 4100,
+        },
+        optionId: "short",
+        locale: "en",
+        expectedFormat: "in 1 hr. in 8 min. in 20 sec."
+    },
```

Note that `ms` components will be silently dropped. E.g.
```typescript
{
  input: <DateTimeComponents>{ ms: -1, },
  optionId: "short",
  locale: "en",
  expectedFormat: "1 ms. ago"
},
```
will fail, producing the empty string instead.

---

Milliseconds are not supported by `IntlDurationFormatValueConverter`.

---

Fair enough. Should probably add a sentence in the readme: "Milliseconds will be dropped", or some such.

---

Thats not true, because milliseconds could overflow to seconds.

---

Mmh. True. Badly formulated. "Remaining milliseconds will be dropped"?